doc/model.rst

```diff
@@ -0,0 +1,932 @@
+
+When the ``Model`` is started via ``Experiment.start()``, the TF function will be loaded to the
+standard ``Orchestrator``. The function can then be executed on the ``Orhcestrator`` via a SmartSim
+client (:ref:`SmartRedis<dead_link>`) within the application code.
+
+.. _TS_from_file:
+------------------------------
+Attach a TorchScript from file
+------------------------------
+Users can attach TorchScript functions from a file to a ``Model`` and upload them to a
+colocated or standard ``Orchestrator``. This functionality is supported by the ``Model.add_script()``
+function's `script_path` parameter. The function supports
+flexible device selection, allowing users to choose between `"GPU"` or `"CPU"` via the `device` parameter.
+In environments with multiple devices, specific device numbers can be specified using the
+`devices_per_node` parameter.
+
+When specifying a TorchScript using ``Model.add_script()``, the
+following arguments are offered:
+
+- `name` (str): key to store script under.
+- `script` (t.Optional[str] = None): TorchScript code (only supported for non-colocated orchestrators).
+- `script_path` (t.Optional[str] = None): path to TorchScript code.
+- `device` (t.Literal["CPU", "GPU"] = "CPU"): device for script execution, defaults to “CPU”.
+- `devices_per_node` (int = 1): The number of GPU devices available on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- `first_device` (int = 0): The first GPU device to use on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+
+Continue to the :ref:`Loading a TorchScript from File<TS_from_file_ex>` example.
+
+.. _TS_from_file_ex:
+Example: Loading a TorchScript from File
+----------------------------------------
+This example walks through the steps of instructing SmartSim to load a TorchScript from file
+to a ``Orchestrator``.
+
+.. note::
+    This example assumes:
+
+    - a ``Orchestrator`` is launched prior to the ``Models`` execution
+    - an initialized ``Model`` named `smartsim_model` exists within the ``Experiment`` workflow
+
+**Define a TorchScript script**
+
+For the example, we create the create the Python script `torchscript.py`. The file contains a
+simple torch function shown below:
+
+.. code-block:: python
+
+    def negate(x):
+        return torch.neg(x)
+
+**Attach the TorchScript script to a SmartSim Model**
+
+Assuming an initialized ``Model`` named `smartsim_model` exists, we add a TorchScript script using
+the ``Model.add_script()`` function and specify the script path to the parameter `script_path`:
+
+.. code-block:: python
+
+    smartsim_model.add_script(name="example_script", script_path="path/to/torchscript.py", device="GPU", devices_per_node=2, first_device=0)
+
+In the above ``smartsim_model.add_script()`` code snippet, we offer the following arguments:
+
+-  `name` ("example_script"): key to store script under.
```

Comment:
key to store function under.

->

reference name for the script inside of the ``Orchestrator``